File: fortran-tests/gemm-11.f90
subroutine gemm(m, n, k, alpha, A, lda, B, ldb, C, ldc)
  integer :: m, n, k
  integer :: lda, ldb, ldc
  real, dimension(m * n) :: C
  real, dimension(m * k) :: A
  real, dimension(k * n) :: B
  real :: alpha
  real :: aa, bb

  do mm = 1, m
    do nn = 1, n
      do i = 1, k
        aa = A(mm + i * lda)
        bb = B(nn + i * ldb)
        C(mm + nn * ldc) = C(mm + nn * ldc) + alpha * aa * bb
      end do
    end do
  end do
end subroutine

File: cplus-tests/gemm-11.cc
void basicSgemm(int m, int n, int k, float alpha, const float *A, int lda, const float *B, int ldb, float *C, int ldc )
{
  for (int mm = 0; mm < m; ++mm) {
    for (int nn = 0; nn < n; ++nn) {
      for (int i = 0; i < k; ++i) {
        float a = A[mm + i * lda];
        float b = B[nn + i * ldb];
        C[mm+nn*ldc] += alpha * a * b;
      }
    }
  }
}
